staging/src/k8s.io/apiserver/pkg/server/options/admission.go

```diff
@@ -151,41 +157,54 @@ func (a *AdmissionOptions) Validate() []error {
 	errs := []error{}
 
 	registeredPlugins := sets.NewString(a.Plugins.Registered()...)
-	for _, name := range a.PluginNames {
+	for _, name := range a.EnablePlugins {
 		if !registeredPlugins.Has(name) {
-			errs = append(errs, fmt.Errorf("admission-control plugin %q is invalid", name))
+			errs = append(errs, fmt.Errorf("enable-admission-plugins plugin %q is unknown", name))
 		}
 	}
 
+	for _, name := range a.DisablePlugins {
+		if !registeredPlugins.Has(name) {
+			errs = append(errs, fmt.Errorf("disable-admission-plugins plugin %q is unknown", name))
+		}
+	}
+
+	enablePluginSet := sets.NewString(a.EnablePlugins...)
+	disablePluginSet := sets.NewString(a.DisablePlugins...)
+	if len(enablePluginSet.Intersection(disablePluginSet).List()) > 0 {
+		errs = append(errs, fmt.Errorf("enable-admission-plugins %q and disable-admission-plugins %q "+
+			"must not contain same plugins", a.EnablePlugins, a.DisablePlugins))
+	}
+
+	// Verify RecommendedPluginOrder.
+	if sets.NewString(a.RecommendedPluginOrder...).Len() != len(a.Plugins.Registered()) {
```

Comment:
actually compare to make sure the sets are equal and report the offenders

Reply:
OK, more effective check.